app/models/publication.rb

```diff
@@ -50,13 +50,14 @@ def build_or_update(author, contribution_hash = {})
       # Ensure the contribution attributes contain the right author
       contribution_hash[:author_id] = author.id
       contribution_hash[:cap_profile_id] = author.cap_profile_id if author.cap_profile_id.present?
-      contrib = where(author_id: author.id).first_or_initialize
+      contrib = find_or_initialize_by(author_id: author.id) do |new_contrib|
+        new_contrib.assign_attributes(contribution_hash)
+        new_contrib.publication = proxy_association.owner
+      end
```

Comment:
What does `new_contrib.publication = proxy_association.owner` do when the `find_or_initialize_by` returns an existing record and the publication association is already established?